### PR TITLE
[REVIEW] Adding simple support for n_clusters=1 in agglomerative clustering and ignoring self-loops

### PR DESCRIPTION
**Pull request by cjnolet**

### CODE REVIEW

File: cpp/include/raft/sparse/mst/detail/mst_solver_inl.cuh [hunk 1]
```diff
@@ -83,12 +83,12 @@ MST_solver<vertex_t, edge_t, weight_t>::MST_solver(
   //Initially, color holds the vertex id as color
   auto policy = rmm::exec_policy(stream);
   if (initialize_colors_) {
-    thrust::sequence(policy->on(stream), color.begin(), color.end(), 0);
-    thrust::sequence(policy->on(stream), color_index, color_index + v, 0);
+    thrust::sequence(policy, color.begin(), color.end(), 0);
```
Comment (dantegd) [inline cpp/include/raft/sparse/mst/detail/mst_solver_inl.cuh:86]: The errors in CI in this test seem to be coming from a missing 

```C++
#include <rmm/exec_policy.hpp>
```

in this file

File: cpp/include/raft/sparse/hierarchy/detail/connectivities.cuh [hunk 2]
```diff
@@ -71,10 +73,25 @@ struct distance_graph_impl<raft::hierarchy::LinkageDistance::KNN_GRAPH,
...
+        return (self_loop * std::numeric_limits<value_t>::max()) +
+               (!self_loop * thrust::get<2>(tup));
+      });
+
     raft::sparse::convert::sorted_coo_to_csr(knn_graph_coo.rows(),
                                              knn_graph_coo.nnz, indptr.data(),
                                              m + 1, d_alloc, stream);
 
+    // TODO: Wouldn't need to copy here if we could compute knn
+    // graph directly on the device uvectors
```
Comment (dantegd) [inline cpp/include/raft/sparse/hierarchy/detail/connectivities.cuh:94]: Is this todo captured in an issue?

**Review by dantegd — CHANGES_REQUESTED** [review]
Finished the review, changes looks good, just had a request and the prior suggestion of what's probably wrong with the CI failures

File: cpp/include/raft/sparse/mst/detail/mst_solver_inl.cuh (hunk 1 above)
Comment (cjnolet) [reply cpp/include/raft/sparse/mst/detail/mst_solver_inl.cuh:86]: @divyegala cleaned this up in a different PR for HDBSCAN. I'm guessing the root cause for some of the thrust compilation issues we've been seeing downstream is that MST is some old old API calls. @divye what do you think about cherry-picking that commit from our HDBSCAN branch into a new PR?

**Review by dantegd — APPROVED** [review]

**Comment by dantegd** [comment]
@gpucibot merge

**Pull request merged**